application-admintools-default/src/main/java/com/xwiki/admintools/internal/health/checks/configuration/ConfigurationJavaHealthCheck.java

```diff
@@ -0,0 +1,99 @@
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal.health.checks.configuration;
+
+import java.util.Map;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+
+import com.xwiki.admintools.health.HealthCheck;
+import com.xwiki.admintools.health.HealthCheckResult;
+import com.xwiki.admintools.health.HealthCheckResultLevel;
+
+/**
+ * Implementation of {@link HealthCheck} for checking the Java configuration.
+ *
+ * @version $Id$
+ */
+@Component
+@Named(ConfigurationJavaHealthCheck.HINT)
+@Singleton
+public class ConfigurationJavaHealthCheck extends AbstractConfigurationHealthCheck
+{
+    /**
+     * Component identifier.
+     */
+    public static final String HINT = "configurationJava";
+
+    @Override
+    public HealthCheckResult check()
+    {
+        Map<String, String> configurationJson = getConfigurationProviderJSON();
+        String javaVersionString = configurationJson.get("javaVersion");
+        if (javaVersionString == null) {
+            logger.warn("Java version not found!");
+            return new HealthCheckResult("adminTools.dashboard.healthcheck.java.warn", HealthCheckResultLevel.WARN);
+        }
+        String xwikiVersionString = configurationJson.get("xwikiVersion");
+        float xwikiVersion = parseFloat(xwikiVersionString);
+        float javaVersion = parseFloat(javaVersionString);
+        if (isNotJavaXWikiCompatible(xwikiVersion, javaVersion)) {
+            logger.error("Java version is not compatible with the current XWiki installation!");
+            return new HealthCheckResult("adminTools.dashboard.healthcheck.java.error", HealthCheckResultLevel.ERROR,
+                javaVersionString, xwikiVersionString);
+        }
+        return new HealthCheckResult("adminTools.dashboard.healthcheck.java.info", HealthCheckResultLevel.INFO);
+    }
+
+    private static float parseFloat(String javaVersionString)
+    {
+        String[] parts = javaVersionString.split("\\.");
+        return Float.parseFloat(parts[0] + "." + parts[1]);
+    }
+
+    private boolean isNotJavaXWikiCompatible(float xwikiVersion, float javaVersion)
```

Comment:
It's not a best practice to use negation in the variable or method name, because it just adds to the time you need to understand the meaning when it's used (e.g. !isNotCompatible https://softwareengineering.stackexchange.com/a/196837)
In your case, it's also not in line with what the method returns (I see it returns isCompatible). So I suggest to rename the method to `isJavaXWikiCompatible`